src/bus/mod.rs

```diff
@@ -45,11 +43,8 @@ enum Listener {
 }
 
 impl Bus {
-    pub async fn for_address(address: Option<&str>) -> Result<Self> {
-        let mut address = match address {
-            Some(address) => Address::from_str(address)?,
-            None => Address::from_str(&default_address())?,
-        };
+    pub async fn for_address(address: &str) -> Result<Self> {
```

Comment:
I think it should still be the responsibility of this code (the library, rather than the binary) to figure out the default address. This method can get the config (when we've it) passed to it to make this decision if it's given a `None` address.